integration/token/fungible/views/conversion.go

```diff
@@ -0,0 +1,209 @@
+			outputs, err := tx.Outputs()
+			assert.NoError(err, "failed getting outputs")
+			assert.True(outputs.Count() > 0, "expected at least one output")
+			assert.True(outputs.ByRecipient(id).Count() > 0, "expected at least one output assigned to [%s]", id)
+			// TODO: restore this
+			// actualAmount := outputs.ByRecipient(id).Sum().Uint64()
+			// assert.True(actualAmount == amount, "expected outputs to sum to [%d], got [%d]", i.Amount, actualAmount)
+
+			// If everything is fine, the recipient accepts and sends back her signature.
+			// Notice that, a signature from the recipient might or might not be required to make the transaction valid.
+			// This depends on the driver implementation.
+			span.AddEvent("accept_conversion")
+			_, err = context.RunView(ttx.NewAcceptView(tx))
+			assert.NoError(err, "failed to accept new tokens")
+
+			// Before completing, the recipient waits for finality of the transaction
+			span.AddEvent("ask_for_finality")
+			_, err = context.RunView(ttx.NewFinalityView(tx, ttx.WithTimeout(1*time.Minute)))
+			assert.NoError(err, "new tokens were not committed")
+
+			return tx.ID(), nil
+		},
+	))
+}
+
+type ConversionInitiatorViewFactory struct{}
+
+func (p *ConversionInitiatorViewFactory) NewView(in []byte) (view.View, error) {
+	f := &ConversionInitiatorView{Conversion: &Conversion{}}
+	err := json.Unmarshal(in, f.Conversion)
+	assert.NoError(err, "failed unmarshalling input")
+
+	return f, nil
+}
+
+type ConversionResponderView struct {
+	Auditor string
+}
+
+func (p *ConversionResponderView) Call(context view.Context) (interface{}, error) {
+	span := context.StartSpan("conversion_responder_view")
+	defer span.End()
+	// First the issuer receives the conversion request
+	conversionRequest, err := ttx.ReceiveConversionRequest(context)
+	assert.NoError(err, "failed to receive conversion request")
+
+	// Now we have an inversion of roles. The issuer becomes an initiator.
+	// This is a trick to reuse the code used in IssueCashView
+	return context.RunView(nil, view.AsInitiator(), view.WithViewCall(func(context view.Context) (interface{}, error) {
+		// Before assembling the transaction, the issuer can perform any activity that best fits the business process.
+		// In this example, if the token type is USD, the issuer checks that no more than 230 units of USD
+		// have been issued already including the current request.
+		// No check is performed for other types.
+		wallet := token.GetManagementService(context, token.WithTMSID(conversionRequest.TMSID)).WalletManager().IssuerWallet("")
+		assert.NotNil(wallet, "issuer wallet not found")
+
+		// At this point, the issuer is ready to prepare the token transaction.
+		// The issuer creates a new token transaction and specifies the auditor that must be contacted to approve the operation.
+		var tx *ttx.Transaction
+		var auditorID string
+		if len(p.Auditor) == 0 {
```

Comment:
We don't need to check whether the tx is unspendable here, right?

Reply:
you mean the tokens? It is up to the issuer to perform all the checks that are relevant